From: Sven Mueller <debian@incase.de>
Date: Tue, 4 Aug 2015 10:38:05 +0200
Subject: Fix syslog prefix

Make sure all programs log (to syslog) with "cyrus/<program>" as the
log prefix.

@DPATCH@
---
 imap/global.c       | 7 ++++++-
 ptclient/ptexpire.c | 2 +-
 ptclient/test.c     | 2 +-
 ptclient/test2.c    | 2 +-
 4 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/imap/global.c b/imap/global.c
index bfe081c..b038d3d 100644
--- a/imap/global.c
+++ b/imap/global.c
@@ -148,6 +148,9 @@ static int get_facility(const char *name)
     return SYSLOG_FACILITY;
 }
 
+/* syslog prefix tag */
+static char syslog_prefix[20];
+
 /* Called before a cyrus application starts (but after command line parameters
  * are read) */
 EXPORTED int cyrus_init(const char *alt_config, const char *ident, unsigned flags, int config_need_data)
@@ -181,7 +184,9 @@ EXPORTED int cyrus_init(const char *alt_config, const char *ident, unsigned flag
     
     /* xxx we lose here since we can't have the prefix until we load the
      * config file */
-    openlog(config_ident, syslog_opts, SYSLOG_FACILITY);
+    strncpy(syslog_prefix, "cyrus/", sizeof(syslog_prefix));
+    strncat(syslog_prefix, config_ident, sizeof(syslog_prefix) - 7);
+    openlog(syslog_prefix, syslog_opts, SYSLOG_FACILITY);
 
     /* Load configuration file.  This will set config_dir when it finds it */
     config_read(alt_config, config_need_data);
diff --git a/ptclient/ptexpire.c b/ptclient/ptexpire.c
index 86358a3..940e115 100644
--- a/ptclient/ptexpire.c
+++ b/ptclient/ptexpire.c
@@ -113,7 +113,7 @@ int main(int argc, char *argv[])
 	fatal("must run as the Cyrus user", EC_USAGE);
     }
     
-    openlog("ptexpire", LOG_PID, SYSLOG_FACILITY);
+    openlog("cyrus/ptexpire", LOG_PID, SYSLOG_FACILITY);
 
     while ((opt = getopt(argc, argv, "C:E:")) != EOF) {
 	switch (opt) {
diff --git a/ptclient/test.c b/ptclient/test.c
index 920fe35..a0da9e6 100644
--- a/ptclient/test.c
+++ b/ptclient/test.c
@@ -58,7 +58,7 @@ int main(int argc, char **argv) {
     cacheid=cache;
   } else
     cacheid=NULL;
-  openlog("pttest", LOG_PID, SYSLOG_FACILITY);  
+  openlog("cyrus/pttest", LOG_PID, SYSLOG_FACILITY);  
   
   if (!auth_setid(argv[1],cacheid))
     printf ("Auth_memberof(%s,%s) is %d\n", argv[1], argv[2],
diff --git a/ptclient/test2.c b/ptclient/test2.c
index 84e2fbf..b3ba184 100644
--- a/ptclient/test2.c
+++ b/ptclient/test2.c
@@ -45,7 +45,7 @@
 
 int main(void) {
   char cacheid[16]="4224423";
-  openlog("testr", LOG_PID, SYSLOG_FACILITY);
+  openlog("cyrus/testr", LOG_PID, SYSLOG_FACILITY);
   
   if (!auth_setid("cg2v@club.cc.cmu.edu",cacheid))
     printf ("Auth_memberof(cg2v,cg2v:me) is %d\n",
